kafka/src/main/java/io/micronaut/configuration/kafka/intercept/KafkaClientIntroductionAdvice.java

```diff
@@ -340,22 +337,25 @@ public void onComplete() {
                     LOG.trace("Beginning transaction for producer: {}", producerState.transactionalId);
                     kafkaProducer.beginTransaction();
                 }
+
+                LOG.trace("Sending record {} with producer {}", record, kafkaProducer);
```

Comment:
I may have outdated info in my brain about this, but I thought the only time a LOG.isXEnabled check is really needed in order to avoid an unnecessary performance hit is when using inline String concatenation instead of the built in String interpolation?

Reply:
It's probably me that's outdated... I just remember getting a PR rejected as I didn't use the conditional check (but prior to working on Micronaut, I always followed what you're saying)

@sdelamo do we have a rule with this?

Reply:
This seems to confirm my recollection: https://www.slf4j.org/faq.html#logging_performance

If we want the absolute fastest no-op log statements, than it would seem using isXEnabled would be justified when using the built-in interpolation with 3 or more parameters, and definitely still when using manual concatenation, but otherwise using the interpolation is equivalent in performance.